pydantic/main.py

```diff
@@ -484,13 +484,18 @@ def model_construct(cls: type[Model], _fields_set: set[str] | None = None, **val
         """
         m = cls.__new__(cls)
         fields_values: dict[str, Any] = {}
+        defaults: dict[str, Any] = {}  # keeping this separate from `fields_values` helps us compute `_fields_set`
         for name, field in cls.model_fields.items():
             if field.alias and field.alias in values:
-                fields_values[name] = values[field.alias]
+                fields_values[name] = values.pop(field.alias)
             elif name in values:
-                fields_values[name] = values[name]
+                fields_values[name] = values.pop(name)
             elif not field.is_required():
-                fields_values[name] = field.get_default(call_default_factory=True)
+                defaults[name] = field.get_default(call_default_factory=True)
```

Comment:
Do we care about the actual value here or just that there is a default? Calling the default factory seems suspect

Reply:
We need the actual value as we are using it to populate the dict for the model. `call_default_factory=True` just means that it will be called if present. We would only do this if you don't pass a value for the field, and given that if that were the case and we didn't do this it would result in a malformed model I think this is preferable. Either way, there was an existing and old test for this: `tests.test_construction.test_construct_default_factory`

Reply:
Oh I see this is only when calling model_construct